symbulate/results.py

```diff
@@ -330,61 +365,216 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
             x_height = x_count.values()
             y_height = y_count.values()
             
-            discrete_x = sum([(i > 1) for i in x_height]) > .8 * len(x_height)
-            discrete_y = sum([(i > 1) for i in y_height]) > .8 * len(y_height)
+            discrete_x = discrete_check(x_height)
+            discrete_y = discrete_check(y_height)
 
             if type is None:
-                type = "scatter"
+                type = ("scatter",)
 
             if alpha is None:
                 alpha = .5
 
-            if type == "scatter":
-                if jitter:
-                    x += np.random.normal(loc=0, scale=.01 * (max(x) - min(x)), size=len(x))
-                    y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
-                plt.scatter(x, y, color=color, alpha=alpha, **kwargs)
-            elif type == "tile" and discrete_x and discrete_y:
-                res = pd.DataFrame({'X': x, 'Y': y})
-                res['num'] = 1
-                temp = pd.pivot_table(res, values = 'num', index = ['Y'],
-                    columns = ['X'], aggfunc = np.sum)
-                temp = temp / len(x)
-                sns.set()
-                sns.cubehelix_palette(8)
-                fig, ax = plt.subplots(1, 1)
-                sns.heatmap(temp, ax=ax, linewidths = 0.03).invert_yaxis()
-                ax.set_ylabel('')
-                ax.set_xlabel('')
-            elif type == "mosaic" and discrete_x and discrete_y:
-                res = pd.DataFrame({'X': x, 'Y': y})
-                ct = pd.crosstab(res['Y'], res['X'])
-                ctplus = ct + 1e-8
-                labels = lambda k: ""
-                fig, ax = plt.subplots(1, 1)
-                temp = mosaic(ctplus.unstack(), ax = ax, labelizer = labels, axes_label = False)
-            elif discrete_x and discrete_y:
-                raise Exception("Must have type='mosaic', 'tile', or 'scatter' if discrete.")
-            elif discrete_x == True and discrete_y == False and type == 'violin':
-                fig, ax = plt.subplots(1, 1)
-                res = pd.DataFrame({'X': x, 'Y': y})
-                sns.violinplot(x='X', y='Y', data=res)
-                ax.set_ylabel('')
-                ax.set_xlabel('')
-            elif discrete_x == False and discrete_y == True and type == 'violin':
-                #TODO change this situation based on Dr. Ross's suggestions
+            if bins is None:
+                if 'mixed-tile' in type:
+                    bins = 10
+                else:
+                    bins = 30
+
+            if 'tile' in type and (not discrete_x or not discrete_y):
+                print('type=\'tile\' is only valid for 2 discrete variables')
```

Comment:
If you used double quotes " ", you wouldn't need to escape the single quotes ' ' inside the string.